### ISSUE TITLE
VoiceOver detects the elements behind an open drawer as if they are on screen

### ISSUE THREAD

**Issue by StefanoSarioli** [react-community/react-navigation#311]
Hi everyone.
I was evaluating the accessibility of this library for a possible work when I found out this problem.
If I use VoiceOver with a Drawer navigator everything is fine till the drawer is close, but if I open the Drawer and try to scroll trough the menu elements using VoiceOver the app consider the content of the screen actually hidden by the drawer as valid elements and scroll trough them too.
For more detail I've opened a [question](https://stackoverflow.com/questions/42159122/making-the-background-screen-beneath-the-open-drawer-with-react-navigation-und) on Stack Overflow.

I know that even on native iOS development to prevent this effect you need to call a particular method cause Drawer navigation is not a native component, I'm not really good on native iOS development so maybe I'm wrong here.
I think that react navigation should have the expected behavior with VoiceOVer as default otherwise no app written in it, that use a Drawer navigator, will be accessible by default. 

I've started investigating possible solutions but I'm running out of ideas and at the moment nothing is working as expected, so even just a suggestion on how to deal with this problem will be wonderful.

**Comment by ericvicenti** [comment]
Yikes, we need to work on this. Hopefully RN's existing [accessibility API](https://facebook.github.io/react-native/docs/accessibility.html) will be sufficient to fix this.

**Comment by StefanoSarioli** [comment]
Hi, I'm sorry to have disappeared but I was working on the problem.
I found out that this is fixable by exposing a new property for the View component and using it on the react-native-drawer-layout component.
Now I'm working on the PR for the react-native-drawer-layout component and waiting for the implementation of the property on the View component.
As soon as the PR will be accepted we could close this issue.

**Comment by mohamadkhan19** [comment]
Solution for this issue is to set "accessibilityViewIsModal" property to "true" for the Parent "View" component for each Container.

"...a window that contains sibling views A and B, setting accessibilityViewIsModal to true on view B causes VoiceOver to ignore the elements in the view A. "
Ref: https://facebook.github.io/react-native/docs/accessibility.html

Eg:

render() {
        return (
            <View accessibilityViewIsModal={true}>
                {/* some code. */}
            </View>
        );
    }

**Comment by ArslanAbakarov** [comment]
Guys, I don't get how to use accessibilityViewIsModal.
I've added it to every parent View and it still reads behind the scene elements.

**Comment by DimensionSlider** [comment]
Hi guys,
I don't get how to use it. Can any one provide an example please?
I've added accessibilityViewIsModal to every parent View and VoiceOver still reads out behind the scenes stuff.

**Comment by mattmallon** [comment]
Any updates on this? Using the most recent version of React Navigation (3.9.1) and iOS voiceover, the contents of the screen behind the drawer are read but not the contents of the drawer itself unless the user taps inside the drawer. I tried adding `accessibilityViewIsModal` to the component I'm specifying in my login drawer config's`contentComponent` option but it did not have the intended effect. This seems to be something that should be specified on the side of React Navigation rather than on the developer's side of the house when specifying the contentComponent. 

I noticed that there is a value for `accessibilityViewIsModal` included in the drawer navigation at the following line of code in github, but it doesn't seem to have the intended effect after a call has been made to `toggleDrawer()`: https://github.com/react-navigation/drawer/blob/master/src/views/Drawer.tsx#L544

**Comment by pjivers-bom** [comment]
+1 waiting for an update on this issue

**Comment by annieneedscoffee** [comment]
+ 1 
I've done kind of a work around by changing state on each screen to menuOpen: true whenever my drawer open button is pressed and onDidBlur, and to menuOpen: false onWillFocus, then using accessible={!this.state.menuOpen} in components, but this gets really messy with nested navigators. It's not a fully working solution.

**Comment by zabojad** [comment]
Currently having similar issues with `@react-navigation/drawer@6.4.1`... The content of the drawer gets read by voice over when the drawer is closed :(...
